src/animation_player.rs

```diff
@@ -98,44 +116,80 @@ pub fn match_bones(
 
                             commands.entity(entity).insert(BoneIndex(joint_index));
 
-                            let mut offset_x = joint.data().offset().x;
-                            let mut offset_y = joint.data().offset().y;
-                            let mut offset_z = joint.data().offset().z;
+                            let mut current_offset_x = joint.data().offset().x;
+                            let mut current_offset_y = joint.data().offset().y;
+                            let mut current_offset_z = joint.data().offset().z;
 
-                            let rotation0;
-                            let rotation1;
-                            let rotation2;
+                            let mut next_offset_x = joint.data().offset().x;
+                            let mut next_offset_y = joint.data().offset().y;
+                            let mut next_offset_z = joint.data().offset().z;
+
+                            let current_rotation0: f32;
+                            let current_rotation1: f32;
+                            let current_rotation2: f32;
+
+                            let next_rotation0: f32;
+                            let next_rotation1: f32;
+                            let next_rotation2: f32;
```

Comment:
maybe use [`Vec3`](https://docs.rs/bevy/latest/bevy/math/struct.Vec3.html) instead of 3 `floats`.
e.g.
```rust
let offset = joint.data().offset();
let mut current_offset = Vec3::new(offset.x, offset.y, offset.z);
```